### ISSUE TITLE
Excel, selected cells not read when choosing cells in a formula

### ISSUE THREAD

**Issue by Qchristensen** [nvaccess/nvda#8411]
In Excel, if you press = then press the arrow keys, the focus moves to select other cells to add to your formula.  NVDA does not read this.  eg:
1. In cell, A1, type =
2. press the right arrow. the focus moves to B1, the contents of A1 become "=B1", and NVDA reads "Blank".
3. Press down arrow, the focus moves to B2, and NVDA often does read it when it changes row like this.
4. Press SHIFT+RIGHT ARROW, B2:B3 are selected but this time NVDA reads nothing.

In steps 2, 3, and 4, it would be beneficial to have a way to have the current selection automatically read.   Reading the current line (NVDA+up arrow or in laptop keyboard layout, NVDA+L), reads the cells.  Reading the current focus with NVDA+TAB also works, however there does not seem to be a way to have this read automatically.

Another option is, while editing a cell, press NVDA+number row 7 to set focus moves navigator object off, then press numpad8 (NVDA+SHIFT+.) to read the current review line at any point after that to read the formula bar.

Note that for best results, the Excel option "Allow editing directly in cells" needs to be disabled, as when enabled, reading the contents of any cell while editing does not currently work.  To turn this option off, press ALT+F to open the File menu, then T for Options.  Press CONTROL+TAB until "Advanced options for working with Excel" is selected, then alt+e to toggle "allow editing directly in cells".  press alt+e again if needed to turn off, and when happy, press ENTER to save settings.

**Comment by Adriani90** [comment]
@Qchristensen could you please edit your issue description? Editing directly in cells is now supported again. Just to avoid confusions. :-)

**Comment by Qchristensen** [comment]
Thanks @Adriani90 I've edited the description to remove that as you are correct.  A quick work around for anyone else reading this - you can have the current formula read with NVDA+up arrow (in laptop keyboard layout, press NVDA+L).  The issue as initially presented still exists in NVDA 2019.1.1 and Excel from Office 365 version 16.0.11601.20174.

**Comment by Adriani90** [comment]
Steps 2 and 3 are supported now, although some how buggy since the columns and rows are not reported until the end. Depending on how much cells you move, at some point NVDA just reports "form value" but not the coordinates.

Step 4 does only report "form value" but no coordinates.
Tested with NVDA 2023.1 on MS Excel 365.
